sonarts-core/src/utils/navigation.ts

```diff
@@ -136,6 +136,14 @@ function isToken(node: ts.Node): boolean {
   return node.kind <= ts.SyntaxKind.OfKeyword;
 }
 
+export function isIntersectionTypeNode(node: ts.Node): node is ts.IntersectionTypeNode {
```

Comment:
@stas-vilchik will we create a function for every node? IMO there is no need for that

Reply:
???